shoop/admin/modules/orders/views/create.py

```diff
@@ -86,28 +144,84 @@ def handle_product_data(self, request):
         product_id = request.GET["id"]
         shop_id = request.GET["shop_id"]
         customer_id = request.GET.get("customer_id")
+        quantity = decimal.Decimal(request.GET.get("quantity", 1))
         product = Product.objects.get(pk=product_id)
+        if not product:
+            return {"errorText": _("Product %s does not exist.") % product_id}
         shop = Shop.objects.get(pk=shop_id)
-        ctx_request = RequestFactory().get("/")
-        ctx_request.shop = shop
-        ctx_request.customer = Contact.objects.filter(pk=customer_id).first()
-        ctx_request.user = AnonymousUser()
-        context = get_pricing_module().get_context_from_request(ctx_request)
-        price_info = product.get_price_info(context, quantity=1)
+        try:
+            shop_product = product.get_shop_instance(shop)
+        except ObjectDoesNotExist:
+            return {"errorText": _("Product %(product)s is not available in the %(shop)s shop.") % {
+                "product": product.name, "shop": shop.name}}
+
+        min_quantity = shop_product.minimum_purchase_quantity
+        # Make quantity to be at least minimum quantity
+        quantity = min_quantity if quantity < min_quantity else quantity
+        customer = Contact.objects.filter(pk=customer_id).first() if customer_id else None
+        price_info = get_price_info(shop, customer, product, quantity)
+        error_text = None
+        supplier = shop_product.suppliers.first()  # TODO: Allow setting a supplier?
+        for message in shop_product.get_orderability_errors(
+                supplier=supplier, quantity=quantity, customer=customer, ignore_minimum=True):
+            error_text = str(message.args[0])
         return {
             "id": product.id,
             "sku": product.sku,
             "name": product.name,
+            "quantity": quantity,
+            "purchaseMultiple": shop_product.purchase_multiple,
+            "errorText": error_text,
             "taxClass": {
                 "id": product.tax_class.id,
                 "name": force_text(product.tax_class),
             },
+            "baseUnitPrice": {
+                "value": price_info.base_price.value,
+                "includesTax": price_info.base_price.includes_tax
+            },
             "unitPrice": {
-                "value": price_info.price.value,  # TODO: This is always zero?!
-                "includesTax": price_info.price.includes_tax
+                "value": price_info.price.value,
+                "includesTax": price_info.base_price.includes_tax
             }
         }
 
+    def handle_customer_data(self, request):
+        customer = Contact.objects.filter(pk=request.GET["id"]).first()
+        return {
+            "id": customer.id,
+            "name": customer.name,
+            "isCompany": True if isinstance(customer, CompanyContact) else False,
```

Comment:
No need for `True if x else False`; either just `x` (or maybe preferably `bool(x)`)